README.md

```diff
@@ -1,289 +1,590 @@
+
+| Name | Requirements
+| ---- | ------------
+| `host` | A DNS-resolvable host name or IP address
+| `port` | A valid port number
+| `uri` | A valid URI as defined by [RFC3986](https://tools.ietf.org/html/rfc3986)
+| `username` | A string-based username credential
+| `password` | A string-based password credential
+| `certificates` | A collection of PEM-encoded X.509 certificates, representing a certificate chain used in mTLS client authentication
+| `privateKey` | A PEM-encoded private key used in mTLS client authentication
+
+`Secret` entries that do not meet these requirements **MUST** use different entry names.
+
+# Application Projection
+
+A Binding `Secret` **MUST** be volume mounted into a container at in `$SERVICE_BINDINGS_ROOT/<binding-name>` with directory names matching the name of the binding.  Binding names **MUST** match `[a-z0-9\-\.]{1,253}`.  The `$SERVICE_BINDINGS_ROOT` environment variable **MUST** be declared and can point to any valid file system location.
+
+The `Secret` **MUST** contain a `kind` entry with a value that identifies the abstract classification of the binding.  It is **RECOMMENDED** that the `Secret` also contain a `provider` entry with a value that identifies the provider of the binding.  The `Secret` **MAY** contain any other entry.
+
+The contents of a secret entry may be anything representable as bytes on the file system including, but not limited to, a literal string value (e.g. `db-password`), a language-specific binary (e.g. a Java `KeyStore` with a private key and X.509 certificate), or an indirect pointer to another system for value resolution (e.g. `vault://production-database/password`).
+
+The collection of files within the directory **MAY** change between container launches.  The collection of files within the directory **MUST NOT** change during the lifetime of the container.
+
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+│   ├── kind
+│   ├── connection-count
+│   ├── uri
+│   ├── certificates
+│   └── private-key
+```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into to the application.
```

Comment:
@arturdzm @nebhale Curious, why did we replace multiple services  ( `.spec.services` with `.spec.service` ) ?

Reply:
Ah I read further `SyntheticProvisionedService`  :)

Reply:
the intent is that multiple ServiceBinding resources can target the same application

Reply:
This is something we need to discuss more.  😄   I strongly feel that `ServiceBinding` CR should allow multiple services.  If I have a microservice and I need to connect to 3 services, I shouldn't need to produce 3 different `ServiceBinding` CRs, specially when the subject / target app is exactly the same.

Reply:
@sbose78 Again, for interoperability you define a very primitive contract; one application, one service.  Then you define things on the edges of that contract, like `SyntheticProvidedService` that transform more complex use-cases into that simple kernel of a contract.  It allows a clean, interoperable implementation of the spec, that you can then compose with other implementations from other vendors that add useful, but non-standard behaviors.